flink-connector-kafka/src/main/java/org/apache/flink/connector/kafka/sink/TwoPhaseCommittingStatefulSink.java

```diff
@@ -0,0 +1,34 @@
+package org.apache.flink.connector.kafka.sink;
+
+import org.apache.flink.annotation.Internal;
+import org.apache.flink.api.connector.sink2.StatefulSink;
+import org.apache.flink.api.connector.sink2.TwoPhaseCommittingSink;
+
+import java.io.IOException;
+import java.util.Collection;
+
+/**
+ * A combination of {@link TwoPhaseCommittingSink} and {@link StatefulSink}.
+ *
+ * <p>The purpose of this interface is to be able to pass an interface rather than a {@link
+ * KafkaSink} implementation into the reducing sink which simplifies unit testing.
+ *
+ * @param <InputT> The type of the sink's input
+ * @param <WriterStateT> The type of the sink writer's state
+ * @param <CommT> The type of the committables.
+ */
+@Internal
+public interface TwoPhaseCommittingStatefulSink<InputT, WriterStateT, CommT>
```

Comment:
I'm a bit torn whether or not we should just accept this awkwardness and move on. I understand the need for this "tag" mixin interface, but it just feels really odd.

Interface mixins in Java really doesn't play well with the decorator pattern 😕